### ISSUE TITLE
CannotPullContainerError: error pulling image configuration: toomanyrequests

### ISSUE THREAD

**Issue by lorengordon** [aws-amplify/amplify-console#837]
**Is your feature request related to a problem? Please describe.**

When using the custom image feature to pull from dockerhub, how do we specify a username/access token so we do not get rate limited? Here's the error message:

```
[BUILD_CONTAINER_UNABLE_TO_PULL_IMAGE: Unable to pull customer's container image. CannotPullContainerError: error pulling image configuration: toomanyrequests: Too Many Requests. Please see https://docs.docker.com/docker-hub/download-rate-limit/]
```

Yes, retrying the build works, but hopefully we can find a better solution.

**Comment by Athena96** [comment]
Hi @lorengordon just to confirm, have you setup the "Custom Build Image Requirements" from this doc?
https://docs.aws.amazon.com/amplify/latest/userguide/custom-build-image.html

**Comment by lorengordon** [comment]
@Athena96 Yes, I was actually testing using a [python3.8-nodejs10 image from dockerhub](https://github.com/aws-amplify/amplify-console/issues/595#issuecomment-655540309). The build image *is* working just fine. It just will occasionally hit that rate limit on the dockerhub API.

**Comment by fmarvez** [comment]
I am also seeing this more frequently when using the default node base image: **node:12.18.2**

**Comment by lukehedger** [comment]
@hsspain Would be good to understand whether [Docker's changes to rate limits](https://www.docker.com/increase-rate-limits) is going to impact Amplify builds?

**Comment by lucashenning** [comment]
+1 
![image](https://user-images.githubusercontent.com/12414494/98705381-a7d1dc80-234b-11eb-97dd-b08ec9f851fe.png)

**Comment by davad** [comment]
I'm periodically getting the same error.

**Comment by kahdojay** [comment]
Hi, this is a known issue with custom images - please track the fix in #1009, closing this one
